### ISSUE TITLE
Request Header Inheritance

### ISSUE THREAD

**Issue by c1moore** [socialradar/batch-request#22]
Similar to how [Google](https://cloud.google.com/storage/docs/json_api/v1/how-tos/batch) handles headers, batch-request should allow individual request headers to be inherited from the batch request.  This makes it easier for libraries like Angular to intercept the batch request right before it is sent and append headers as necessary.  It also allows us to access `HTTPOnly` cookies on the individual requests.

To quote from the link above,
> The HTTP headers for the outer batch request, except for the Content- headers such as Content-Type, apply to every request in the batch. If you specify a given HTTP header in both the outer request and an individual call, then the individual call header's value overrides the outer batch request header's value. The headers for an individual call apply only to that call.

**Comment by victorquinn** [comment]
Totally on board with this improvement! Would happily accept a PR which would do this, as long as it's backwards compatible!

**Comment by c1moore** [comment]
I don't think that could be guaranteed unless a configuration option was added similar `inheritHeaders` and the default was set to `false`.  Setting it to `true` would allow for header inheritance.

**Comment by c1moore** [comment]
Created a PR.
